Makefile.in

```diff
@@ -309,6 +309,10 @@ broken-test-suite:				\
 broken-%-test-suite:
 	@$(MAKE) $(FLAGS) check-$*-test-suite ACTION=broken
 
+check-octave-test-suite: precompile-octave-headers
+precompile-octave-headers:
+	@case "$(ACTION)" in *clean*);; *) cd Examples && $(MAKE) $(FLAGS) precompile-octave-headers;; esac
+
```

Comment:
I don't think this will work if you navigate into the Examples/test-suite/octave directory and run just one of the octave tests, such as `make namespace_class.cpptest`. Take a look at Java testing which requires CommentParser.java to be compiled as a prerequisite step. I suggest looking at using this technique for all the tests.

https://github.com/swig/swig/blob/f5e1e689fc583333a870c071c8e4488cf67dc7ea/Examples/test-suite/java/Makefile.in#L113-L119

Reply:
I've implemented something similar, so the pre-compilation step should work if running the test suite from the top-level, or individual tests from `Examples/test-suite/octave`